src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryRecordValue.cs

```diff
@@ -41,5 +43,18 @@ protected override bool TryGetField(FormulaType fieldType, string fieldName, out
         {
             return _fields.TryGetValue(fieldName, out result);
         }
+
+        public override void UpdateField(NamedValue newNamedValue)
+        {
+            if (newNamedValue == null || _sourceDict.IsReadOnly)
+            {
+                base.UpdateField(newNamedValue);
+            }
+
+            if (_sourceDict.ContainsKey(newNamedValue.Name))
```

Comment:
> if (_sourceDict.ContainsKey(newNamedValue.Name)) [](http://example.com/codeflow?start=11&length=49)

Why do we do this check?

We should be able to set blank values to non-blank values.

Reply:
From my tests in PA, when you are trying to Patch a base record from data source, the function should not add new fields to the base record. Only fields that exists in the base record will be modified.

Reply:
"add new fields" can mean two things:
1. add a field that is not already on the Type
2. change an existing blank value to non-blank. 

I suspect 1 is not allowed, but I think 2 should be (we can double check).  This will block 2.

Reply:
ClearCollect(_x, {f1:1, f2:2}, {f1:1, f3:3})
IsBlank(First(_x).f3) // true

Patch(_x, First(_x), { f3:33}) // Succeeds, now First(_x).f3 = 33